hack/generator/pkg/astmodel/type_visitor.go

```diff
@@ -67,7 +69,11 @@ func (tv *TypeVisitor) VisitDefinition(td TypeDefinition, ctx interface{}) (*Typ
 
 	name, ok := visitedName.(TypeName)
 	if !ok {
-		return nil, errors.Errorf("expected visit of %q to return TypeName, not %T", td.Name(), visitedName)
+		if t, ok := visitedName.(*TypeName); ok {
+			name = *t
```

Comment:
I think it was only coming as `*TypeName` because of the place where you returned a `*TypeName` (Which I believe I highlighted in a different comment?)

I think if you change that you can drop this too.

Reply:
Yes, it can be removed - I'll do so.